ip-rule: add page (#6504)
tldr-pages/tldr: pages/linux/ip-rule.md

# ip rule

> > IP Routing policy database management.

Reviewer: ```suggestion
> IP routing policy database management.
```

> More information: <https://man7.org/linux/man-pages/man8/ip-rule.8.html>.

- Display the routing policy:

> `ip rule {{show|list}}`

Reviewer: If those are exactly the same I think we should just specify one to avoid confusion

Author: Oh. I based a similar command at https://github.com/tldr-pages/tldr/blob/main/pages/linux/ip-route.md
Do you think I should modify it to show or list ?

Reviewer: No problem, this isn't any fixed rule. Let's just wait for some more opinions

Author: Hey @navarroaxel , What do you think about this ?

Reviewer 2: Hmmm. Normally I'd say pick 1 and stick with, but in this case since the other page is doing it I'd go for consistency.

Author: Okay, I'll keep it as it is for consistency

> - Insert a new rule use the source prefix:

Reviewer: I'm not really understanding this sentence 🤔

Author: If I change it to "Adding a rule to the routing policy database based on the source address of the packet", do you understand?

Reviewer: Ah thanks, now I get it. But I think we can change it a little bit:
```suggestion
- Add a new rule based on the source address of the specified package:
```
Could you please also change the following examples to a similar wording?

Author: Okay

> - Add a new rule based on the source address of the specified package:

Author: "package" means a packet in the Internet

> - Add a new rule based on the source address of the specified package:

Reviewer 2: Ah, I see. In that case, we should be much more explicit here. `package` doesn't really make sense in  this context:
```suggestion
- Add a new rule based on packet source addresses:
```

> - Add a new rule based on the source address of the specified package:

Author: Okay. Thanks @sbrl

`sudo ip rule add from {{192.168.178.2/32}}`

> - Add a new rule based on packet destination addresses

Reviewer 3: ```suggestion
- Add a new rule based on packet destination addresses:
```

> - Add a new rule based on packet destination addresses

Reviewer 3: ```suggestion
- Add a new rule based on packet destination addresses:
```

> - Add a new rule based on the destination address of the specified package:

Reviewer 2: The same change should be applied to these other 3 examples too.

Author: Done

`sudo ip rule add to {{192.168.178.2/32}}`

- Delete a new rule use the source prefix:

`sudo ip rule delete from {{192.168.178.2/32}}`

- Delete a new rule use the destination prefix:

`sudo ip rule delete to {{192.168.178.2/32}}`

> - Dump all the deleted rules:

Reviewer: I think `dump` sounds more like `print to the terminal`, even though it is correct. How about just
```suggestion
- Flush all the deleted rules:
```

Author: Okay, I'll fix it

`ip rule flush`

> - Save all rules to file:

Reviewer: ```suggestion
- Save all rules to a file:
```

Author: Okay, I'll fix it

> `ip rule save > ip-rules.dat`

Reviewer: The `{{ }}` syntax won't be displayed, it's just there to hightlight that part
```suggestion
`ip rule save > {{path/to/ip-rules.dat}}`
```

> - Flush all the deleted rules:

Reviewer: ```suggestion
- Flush all deleted rules:
```

> - Restore all rules from file:

Reviewer: ```suggestion
- Restore all rules from a file:
```

> `ip rule restore < ip-rules.dat`

Reviewer: ```suggestion
`ip rule restore < {{path/to/ip-rules.dat}}`
```

> `ip rule save > {{path/to/ip-rules.dat}}`

Reviewer 3: We use `snake_case` for placeholders:
```suggestion
`ip rule save > {{path/to/ip_rules.dat}}`
```

> `ip rule restore < {{path/to/ip-rules.dat}}`

Reviewer 3: Should this be with `sudo`?
```suggestion
`ip rule restore < {{path/to/ip_rules.dat}}`
```
